code/game/objects/items/robot/items/tools.dm

```diff
@@ -250,73 +274,31 @@
 		/obj/item/bonesetter/cyborg,
 	)
 
-/obj/item/borg/cyborg_omnitool
-	name = "broken cyborg tool arm"
-	desc = "Some internal part of a broken cyborg."
-	icon = 'icons/mob/silicon/robot_items.dmi'
-	icon_state = "lollipop"
-	///Ref to the toolbox, since our own loc will be changing
-	var/obj/item/cyborg_omnitoolbox/toolbox
-	///Ref to currently selected tool, if any
-	var/obj/item/selected
-
-/obj/item/borg/cyborg_omnitool/Initialize(mapload)
-	. = ..()
-	if(!iscyborg(loc.loc))
-		return
-	var/obj/item/robot_model/model = loc
-	var/obj/item/cyborg_omnitoolbox/chassis_toolbox = model.toolbox
-	if(!chassis_toolbox)
-		return
-	toolbox = chassis_toolbox
-	toolbox.omnitools += src
-
-/obj/item/borg/cyborg_omnitool/attack_self(mob/user)
-	var/list/radial_menu_options = list()
-	for(var/obj/item/borgtool in toolbox.contents)
-		radial_menu_options[borgtool] = image(icon = borgtool.icon, icon_state = borgtool.icon_state)
-	var/obj/item/potential_new_tool = show_radial_menu(user, src, radial_menu_options, require_near = TRUE, tooltips = TRUE)
-	if(!potential_new_tool)
-		return ..()
-	if(potential_new_tool == selected)
-		return ..()
-	for(var/obj/item/borg/cyborg_omnitool/coworker in toolbox.omnitools)
-		if(coworker.selected == potential_new_tool)
-			coworker.deselect() //Can I borrow that please
-			break
-	selected = potential_new_tool
-	icon_state = selected.icon_state
-	playsound(src, 'sound/items/change_jaws.ogg', 50, TRUE)
-	return ..()
-
-/obj/item/borg/cyborg_omnitool/proc/deselect()
-	if(!selected)
-		return
-	selected = null
-	icon_state = initial(icon_state)
-	playsound(src, 'sound/items/change_jaws.ogg', 50, TRUE)
+/obj/item/borg/cyborg_omnitool/medical/Initialize(mapload)
+	omni_toolkit = omni_medical_toolbox
 
-/obj/item/borg/cyborg_omnitool/cyborg_unequip()
-	deselect()
-	return ..()
-
-/obj/item/borg/cyborg_omnitool/pre_attack(atom/atom, mob/living/user, params)
-	if(selected)
-		selected.melee_attack_chain(user, atom, params)
-		return TRUE
 	return ..()
 
+//Toolset for engineering cyborgs, this is all of the tools except for the welding tool. since it's quite hard to implement (read:can't be arsed to)
 /obj/item/borg/cyborg_omnitool/engineering
 	name = "engineering omni-toolset"
 	desc = "A set of engineering tools used by cyborgs to conduct various engineering tasks."
+	icon = 'icons/obj/items_cyborg.dmi'
 	icon_state = "toolkit_engiborg"
 
-/obj/item/borg/cyborg_omnitool/medical
-	name = "surgical omni-toolset"
-	desc = "A set of surgical tools used by cyborgs to operate on various surgical operations."
-	icon_state = "toolkit_medborg"
+	///Cached list of tools for the engineering toolkit
+	var/static/list/omni_engineering_toolbox = list(
```

Comment:
Caching this seems rather redundant. 

I would wager that no more than, maybe four engineering cyborgs would exist at any given time - it's certainly not a huge memory hog. 

In fact, all this does is make it harder to work with, both in the code (you have to do the set-in-init shenanigans) and in game (no vv shenanigans)